File: src/test/java/testdata/java/JsonPropertyTestPOJO.java
package testdata.java;


import com.alibaba.fastjson.annotation.JSONField;
import com.fasterxml.jackson.annotation.JsonProperty;

public class JsonPropertyTestPOJO {

    @JsonProperty
    private String userId;

    @JsonProperty("name")
    private String username;

    @JSONField(name = "pass")
    private String password;

}

File: src/test/java/testdata/kotlin/JsonPropertyTestPOJO.kt
package testdata.kotlin

import com.alibaba.fastjson.annotation.JSONField
import com.fasterxml.jackson.annotation.JsonProperty

class JsonPropertyTestPOJO {

    @JsonProperty
    private val userId: String? = null

    @JsonProperty("name")
    private val username: String? = null

    @JSONField(name = "pass")
    private val password: String? = null
}